zebra-chain/src/value_balance.rs

```diff
@@ -127,6 +129,33 @@ where
             orchard: zero,
         }
     }
+
+    /// To byte array
+    pub fn to_bytes(self) -> [u8; 32] {
+        let transparent = self.transparent.to_bytes();
+        let sprout = self.sprout.to_bytes();
+        let sapling = self.sapling.to_bytes();
+        let orchard = self.orchard.to_bytes();
+        match [transparent, sprout, sapling, orchard].concat().try_into() {
+            Ok(bytes) => bytes,
+            _ => unreachable!("should be impossible to get here"),
```

Comment:
I don't think there's an easy way to get a compile-time error here. So we want to write code that makes the panic obvious.

We use `unreachable!` so that the panic error message explains the bug in the code.
Why is this code unreachable?

(`expect` and `unreachable!` do the same thing here, so you can use either one.)